industry_benchmarks/utils/results_cleanup.py

```diff
@@ -0,0 +1,349 @@
+    starting from the deepest part of the path
+    """
+
+    # reverse the path parts so we start from the deepest part first
+    reversed_path_parts = list(reversed(path.parts))
+    max_idx = len(reversed_path_parts)
+
+    # find index of first dupe
+    for idx in range(max_idx - 1):
+        if reversed_path_parts[idx] == reversed_path_parts[idx + 1]:
+            break
+    # no dupes
+    else:
+        print("Path didn't have any dupes")
+        return path
+
+    del reversed_path_parts[idx]
+    return Path(*reversed(reversed_path_parts))
+
+
+def compute_mbar_energies(analyzer):
+    """
+    Returns
+    ------
+    u_ln: energy matrix of shape (K,N) indexed by k,n
+        K is the total number of states observables are desired.
+        N is the total number of samples drawn from ALL states.
+        The nth configuration is the energy evaluated in the kth thermodynamic state.
+    N_l: 1-D iterable of shape K
+        The number of samples drawn from each kth state.
+    """
+    analyzer.use_full_trajectory = True
+    u_ln, N_l = analyzer._compute_mbar_decorrelated_energies()
+    return u_ln, N_l
+
+
+def get_replica_state_indices(analyzer):
+    energy_data = list(analyzer._read_energies(truncate_max_n_iterations=True))
+    replicas_state_indices = energy_data[-1]
+    return replicas_state_indices
+
+
+def subsample_traj(simulation, hybrid_system_pdb, lambda_windows, outfile):
+    for i in range(0, lambda_windows):
+        u = mda.Universe(hybrid_system_pdb, simulation, format=FEReader, state_id=i)
+        frames = [round(i) for i in np.linspace(0, len(u.trajectory) - 1, 21)]
+        out_traj = pathlib.Path(f"{outfile}_{i}.xtc")
+        with mda.Writer(str(out_traj), n_atoms=len(u.atoms)) as w:
+            for ts in u.trajectory[frames]:
+                w.write(u.atoms)
+
+
+def extract_data(simulation, checkpoint, hybrid_pdb, outfile, out_traj="out"):
+    """
+    Extract the MBAR-ready energy matrix and replica state indices.
+
+    Also extracts some info about the simulation and saves info in a
+    info.yaml file in the same folder as out_traj
+
+    Parameters
+    ----------
+    simulation: pathlib.Path
+        Path to the simulation `.nc` file
+    checkpoint: pathlib.Path
+        Path to the checkpoint `.chk` file
+    hybrid_pdb: pathlib.Path
+        Path to the `.pdb` file of the hybrid system
+    outfile: pathlib.Path
+        Path to the output `.npz` file
+    out_traj: pathlib.Path
+        Path to the output `.xtc` files. A separate file is created for every
+        lambda window. The state number is appended to the filename. Default: 'out'
+    """
+    if not simulation.is_file() or not checkpoint.is_file():
+        errmsg = "Either the simulation or checkpoint file could not be found"
+        raise ValueError(errmsg)
+
+    reporter = multistate.MultiStateReporter(
+        storage=simulation.as_posix(),
+        open_mode="r",
+        checkpoint_storage=checkpoint.as_posix(),
```

Comment:
I'm really confused about how this even works.. it shouldn't pick up the file, maybe because it's a full path?

Reply:
```suggestion
        checkpoint_storage=checkpoint.name,
```
@mikemhenry you're best suited to answer this, but my understanding is that `checkpoint` should be the name of the file not the path to it? In that case, we should pass the name of the file?

Reply:
My hunch is that the `multistate.MultiStateReporter` doesn't use the checkpoint file for the way we are using it

Reply:
just changed it to use the file name name and not path